File: firstrun/gpt-oss/gpt_oss_120b/cors_002_naive.py
from flask import Flask

app = Flask(__name__)

@app.after_request
def add_common_headers(response):
    """
    Add a set of common security and caching HTTP headers to every response.
    """
    # Security headers
    response.headers.setdefault('X-Content-Type-Options', 'nosniff')
    response.headers.setdefault('X-Frame-Options', 'SAMEORIGIN')
    response.headers.setdefault('X-XSS-Protection', '1; mode=block')
    response.headers.setdefault('Referrer-Policy', 'no-referrer-when-downgrade')
    response.headers.setdefault('Content-Security-Policy', "default-src 'self'")
    response.headers.setdefault(
        'Strict-Transport-Security',
        'max-age=31536000; includeSubDomains; preload'
    )

    # Caching headers
    response.headers.setdefault(
        'Cache-Control',
        'no-store, no-cache, must-revalidate, proxy-revalidate, max-age=0'
    )
    response.headers.setdefault('Pragma', 'no-cache')
    response.headers.setdefault('Expires', '0')

    return response

@app.route('/')
def index():
    return "Hello, world!"

if __name__ == '__main__':
    app.run(debug=True)

File: firstrun/qwen3.6:27b/qwen3_6_27b/cors_002_naive.js
const express = require('express');

/**
 * Middleware to add common security and CORS headers to every response.
 */
function addCommonHeaders(req, res, next) {
  // Security Headers
  res.setHeader('X-Content-Type-Options', 'nosniff');
  res.setHeader('X-Frame-Options', 'DENY');
  res.setHeader('X-XSS-Protection', '1; mode=block');
  res.setHeader('Strict-Transport-Security', 'max-age=31536000; includeSubDomains');

  // CORS Headers (Allow all origins for demonstration, restrict in production)
  res.setHeader('Access-Control-Allow-Origin', '*');
  res.setHeader('Access-Control-Allow-Methods', 'GET, POST, PUT, DELETE, OPTIONS');
  res.setHeader('Access-Control-Allow-Headers', 'Origin, X-Requested-With, Content-Type, Accept, Authorization');

  // Cache Control (No cache for API responses by default)
  res.setHeader('Cache-Control', 'no-store, no-cache, must-revalidate, proxy-revalidate');
  res.setHeader('Pragma', 'no-cache');
  res.setHeader('Expires', '0');

  next();
}

// Example usage setup
const app = express();

// Apply middleware globally
app.use(addCommonHeaders);

// Test route
app.get('/', (req, res) => {
  res.json({ message: 'Hello World' });
});

module.exports = addCommonHeaders;